balena/base_request.py

```diff
@@ -137,11 +137,11 @@ def __request(self, url, method, params, endpoint, headers=None,
         request_method = methods[method.lower()]
         url = urljoin(endpoint, url)
         if auth:
-            if not api_key:
-                self.__set_authorization(headers)
-            params = self._format_params(params, api_key=api_key, raw_query=raw_query)
+            # api_key will be None if Auth token is found so we can just pass api_key here.
+            self.__set_authorization(headers, api_key)
+            params = self._format_params(params, raw_query=raw_query)
         else:
-            params = self._format_params(params, api_key=None, raw_query=raw_query)
+            params = self._format_params(params, raw_query=raw_query)
```

Comment:
Let's move this line outside of the `if-else` and drop the `else`.
```suggestion
            params = self._format_params(params, raw_query=raw_query)
```

Reply:
Updated!